Test isolation: global config leaks into unit tests

## Problem

Unit tests in `tests/unit/adapters/test_toml_config_provider.py` fail when a global config file exists at `~/.config/ember/config.toml`.

The tests expect default values but the `TomlConfigProvider.load()` method loads the global config, causing assertion failures like:

```
AssertionError: assert 'minilm' == 'local-default-code-embed'
```

## Affected Tests

- `TestLoadMissingConfig::test_load_missing_file_returns_defaults`
- `TestLoadInvalidConfig::test_load_invalid_toml_returns_defaults`
- `TestPartialConfig::test_load_partial_config_uses_defaults_for_missing`

## Root Cause

The `TomlConfigProvider.load()` method checks for global config at `~/.config/ember/config.toml` before falling back to defaults. Tests don't mock or isolate this.

## Proposed Solution

Mock `get_global_config_path()` in affected tests to return a non-existent path, ensuring tests are isolated from user's actual global config.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -27,6 +27,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   - Existing projects continue to work (full project configs still override everything)
 
 ### Fixed
+- **Test isolation: global config no longer leaks into unit tests** (#254)
+  - Config provider tests now mock `get_global_config_path()` to prevent user's `~/.config/ember/config.toml` from affecting test results
+  - Added `no_global_config` fixture for consistent test isolation
+
 - **SqliteVecAdapter now uses correct embedding dimension** (#249)
   - `SqliteVecAdapter` was hardcoded to 768 dimensions (Jina), breaking MiniLM and BGE models which use 384 dimensions
   - `ember find` and `ember search` commands now pass the embedder's dimension to `SqliteVecAdapter`

--- a/tests/unit/adapters/test_toml_config_provider.py
+++ b/tests/unit/adapters/test_toml_config_provider.py
@@ -15,9 +15,19 @@ def provider() -> TomlConfigProvider:
     return TomlConfigProvider()
 
 
-def _mock_no_global_config(tmp_path: Path) -> Path:
-    """Get a nonexistent global config path for test isolation."""
-    return tmp_path / "nonexistent_global" / "config.toml"
+@pytest.fixture
+def no_global_config(tmp_path: Path):
+    """Mock global config path to ensure test isolation from user's actual config.
+
+    Tests that assert default values must use this fixture to avoid reading
+    the user's ~/.config/ember/config.toml which could override defaults.
+    """
+    nonexistent_global = tmp_path / "nonexistent_global" / "config.toml"
+    with patch(
+        "ember.adapters.config.toml_config_provider.get_global_config_path",
+        return_value=nonexistent_global,
+    ):
+        yield nonexistent_global
 
 
 
@@ -117,9 +127,12 @@ class TestLoadMissingConfig:
     """Tests for loading when config file is missing."""
 
     def test_load_missing_file_returns_defaults(
-        self, provider: TomlConfigProvider, tmp_path: Path
+        self, provider: TomlConfigProvider, tmp_path: Path, no_global_config: Path
     ) -> None:
-        """Test that missing config file returns default configuration."""
+        """Test that missing config file returns default configuration.
+
+        Uses no_global_config fixture to isolate from user's actual global config.
+        """
         ember_dir = tmp_path / ".ember"
         ember_dir.mkdir()
         # config.toml does NOT exist
@@ -134,21 +147,16 @@ class TestLoadMissingConfig:
         assert result.model.mode == default.model.mode
 
     def test_load_missing_directory_returns_defaults(
-        self, provider: TomlConfigProvider, tmp_path: Path
+        self, provider: TomlConfigProvider, tmp_path: Path, no_global_config: Path
     ) -> None:
         """Test that missing ember directory returns default configuration.
 
-        This test must mock global config to isolate from user's actual config.
+        Uses no_global_config fixture to isolate from user's actual global config.
         """
         ember_dir = tmp_path / ".ember"
         # Directory does NOT exist
-        nonexistent_global = tmp_path / "nonexistent_global" / "config.toml"
 
-        with patch(
-            "ember.adapters.config.toml_config_provider.get_global_config_path",
-            return_value=nonexistent_global,
-        ):
-            result = provider.load(ember_dir)
+        result = provider.load(ember_dir)
 
         # Should return defaults
         assert isinstance(result, EmberConfig)
@@ -160,9 +168,16 @@ class TestLoadInvalidConfig:
     """Tests for handling invalid configurations."""
 
     def test_load_invalid_toml_returns_defaults(
-        self, provider: TomlConfigProvider, tmp_path: Path, caplog: pytest.LogCaptureFixture
+        self,
+        provider: TomlConfigProvider,
+        tmp_path: Path,
+        no_global_config: Path,
+        caplog: pytest.LogCaptureFixture,
     ) -> None:
-        """Test that invalid TOML syntax returns defaults with warning."""
+        """Test that invalid TOML syntax returns defaults with warning.
+
+        Uses no_global_config fixture to isolate from user's actual global config.
+        """
         ember_dir = tmp_path / ".ember"
         ember_dir.mkdir()
         config_file = ember_dir / "config.toml"
@@ -184,9 +199,16 @@ model = "broken"
         assert "Failed to parse config.toml" in caplog.text
 
     def test_load_config_with_unknown_keys_returns_defaults(
-        self, provider: TomlConfigProvider, tmp_path: Path, caplog: pytest.LogCaptureFixture
+        self,
+        provider: TomlConfigProvider,
+        tmp_path: Path,
+        no_global_config: Path,
+        caplog: pytest.LogCaptureFixture,
     ) -> None:
-        """Test that unknown keys in config fall back to defaults gracefully."""
+        """Test that unknown keys in config fall back to defaults gracefully.
+
+        Uses no_global_config fixture to isolate from user's actual global config.
+        """
         ember_dir = tmp_path / ".ember"
         ember_dir.mkdir()
         config_file = ember_dir / "config.toml"
@@ -214,9 +236,12 @@ class TestPartialConfig:
     """Tests for partial configurations (missing sections)."""
 
     def test_load_partial_config_uses_defaults_for_missing(
-        self, provider: TomlConfigProvider, tmp_path: Path
+        self, provider: TomlConfigProvider, tmp_path: Path, no_global_config: Path
     ) -> None:
-        """Test that partial config fills in missing sections with defaults."""
+        """Test that partial config fills in missing sections with defaults.
+
+        Uses no_global_config fixture to isolate from user's actual global config.
+        """
         ember_dir = tmp_path / ".ember"
         ember_dir.mkdir()
         config_file = ember_dir / "config.toml"
@@ -239,23 +264,18 @@ topk = 999
         assert result.model.mode == default.model.mode
 
     def test_load_empty_config_returns_defaults(
-        self, provider: TomlConfigProvider, tmp_path: Path
+        self, provider: TomlConfigProvider, tmp_path: Path, no_global_config: Path
     ) -> None:
         """Test that empty config file returns defaults.
 
-        This test must mock global config to isolate from user's actual config.
+        Uses no_global_config fixture to isolate from user's actual global config.
         """
         ember_dir = tmp_path / ".ember"
         ember_dir.mkdir()
         config_file = ember_dir / "config.toml"
         config_file.write_text("")
-        nonexistent_global = tmp_path / "nonexistent_global" / "config.toml"
 
-        with patch(
-            "ember.adapters.config.toml_config_provider.get_global_config_path",
-            return_value=nonexistent_global,
-        ):
-            result = provider.load(ember_dir)
+        result = provider.load(ember_dir)
 
         # Should return defaults (empty config = all defaults)
         default = EmberConfig.default()

